### PR TITLE
[RFR] Use UI-Select for Choice and Choices fields

### PR DESCRIPTION
**Pull request by jpetitcolas**
It adds select filtering and a better UI for multi-selects.

**Before:**
old_anim

**Now:**
anim

It will especially be useful to populate live select lists using AJAX requests.

**Tasks:**
- [x] Implement UI-Select library
- [ ] Prevent ENTER from submitting the form when in select list
- [x] Update or create tests to ensure we use this lib

### CODE REVIEW

**Comment by fzaninotto** [comment]
Promising!

File: README.md [hunk 1]
```diff
@@ -835,6 +835,12 @@ ng-admin has unit tests (powered by karma) and end to end tests (powered by prot
 make test
 ```
 
+**Tip:** if you are working on Karma tests, you can prevent from relaunching the whole process by disabling single-run:
+
+```
+KARMA_SINGLE_RUN=false grunt karma:unit
```
Comment (jpetitcolas) [inline README.md]: Special focus on this line. May save a lot of time!

**Comment by jpetitcolas** [comment]
Miss a last point on this PR: `Prevent ENTER from submitting the form when in select list`. Working on it. As I'm not sure if it depends from `ng-admin` or from `ui-select`, switching this one as RFR.

File: README.md (hunk 1 above)
Comment (manuquentin) [inline README.md]: It can also be changed [here](https://github.com/marmelab/ng-admin/blob/master/Gruntfile.js#L53)

Comment (jpetitcolas) [inline README.md]: It has already been changed. This is not native to Karma, this is specific to `ng-admin`. See [Gruntfile changes](https://github.com/marmelab/ng-admin/pull/505/files#diff-35b4a816e0441e6a375cd925af50752cR53).

**Comment by jpetitcolas** [comment]
Back to WIP: multiple choices required constraint is not taken into account.

File: src/javascripts/ng-admin/Crud/field/maChoicesField.js [hunk 2]
```diff
@@ -1,47 +1,56 @@
...
+        restrict: 'E',
+        compile: function() {
+            return {
+                pre: function(scope, element) {
+                    var field = scope.field();
+                    scope.name = field.name();
+                    scope.v = field.validation();
+
+                    var template = `
+                        <ui-select ${scope.v.required ? 'ui-select-required' : ''} multiple ng-model="$parent.value" ng-required="v.required" id="{{ name }}" name="{{ name }}">
```
Comment (jpetitcolas) [inline src/javascripts/ng-admin/Crud/field/maChoicesField.js]: Not proud of this, but `ng-attr-ui-select-required` did not work.

**Comment by jpetitcolas** [comment]
Back to RFR.

**Pull request merged**